### DISCUSSION TITLE
Boost version

### DISCUSSION THREAD

**Discussion by mickgrig** [valhalla/valhalla#3336] [Q&A] [answered]
Hi, everyone!

Seems like I got new compilation problems after switching to valhalla release 3.1.3 with my boost version 1.69. So, is there some minimum boost version which is recommended to use with valhalla?
Thanks!

**Comment by nilsnolde** [comment, marked as answer]
what's the error message? what module is not working?

officially we support down to [1.51](https://github.com/valhalla/valhalla/blob/aacd79a57629f732ddc09515ae3c60c2cd4e032e/CMakeLists.txt#L222), but nothing in CI checks that, we'd have to build boost from source. current ubuntu 20.04 has v1.71, the vcpkg commit we use for windows CI is even on 1.73. the last I checked in a debian 9 container to build valhalla with boost v1.61 it worked, but that was only the library and python bindings (maybe mjolnir too). that's a few weeks ago already, so could be that smth crept in..

nevertheless it shows again that we should solve #3220 so we can vendor all the header libs and ditch binary boost, that'd solve a few problems.. I started a branch replacing the arg parser but didn't have time lately to continue..

**Reply by nilsnolde** [reply]
Oh and of course your build chain is important: which compiler (and version), which OS and whether you installed boost from a package manager or from an own boost build.

**Reply by mickgrig** [reply]
Build chain is next:
MacOS BigSur (11.5.1), compiler Clang (12.0), boost installed from package manager (version 1.69).

Error message:
"In file included from /usr/local/boost169/include/boost/multi_index/detail/ord_index_impl.hpp:60:
/usr/local/boost169/include/boost/multi_index/detail/ord_index_node.hpp:182:17: error: definition of implicit copy constructor for 'parent_ref' is deprecated because it has a user-declared copy assignment operator [-Werror,-Wdeprecated-copy]"

As far as I could understand, this error is caused by compiler flag -Wextra in the file mjolnir/CMakeLists.txt. And the reason is the absence of copy constructor for the structure "ordered_index_node_compressed_base::parent_ref" which was added in 1.71 boost library.

**Reply by nilsnolde** [reply]
the full error trace would be best, i.e. where in valhalla's code does it happen?

**Reply by mickgrig** [reply]
[trace.txt](https://github.com/valhalla/valhalla/files/7237783/trace.txt)

**Reply by nilsnolde** [reply]
did you resolve it @mickgrig (since you marked it answered)? 

I'm not the very best at interpreting compiler logs, but it looks like it's an internal boost problem right? it seems to be only caused by pulling in some boost headers, not by actual valhalla code. to me that suggests that your boost installation is somehow broken.. brew seems to be [on boost 1.76](https://formulae.brew.sh/formula/boost). can you try that one? are you on apple silicon (M1)?

**Reply by mickgrig** [reply]
Yes, actually I suppose, we can consider this issue as resolved. I think, that it's clearly the boost version problem. Unfortunately, I didn't catch what exact valhalla code causes that error, but if I get some new info about it, I will write here.
And also in our case, unfortunately, it's not so easy to switch our boost-library version and check it.
Processor: 2.3 GHz, Intel Core i9 (8 cores).
Memory: 16 Gb, DDR4
